packages/tpetra/core/src/Tpetra_createDeepCopy_CrsMatrix_def.hpp

```diff
@@ -28,17 +28,23 @@ localDeepCopyFillCompleteCrsMatrix (const CrsMatrix<SC, LO, GO, NT>& A)
   using inds_type = typename local_graph_type::entries_type;
   inds_type ind (view_alloc ("ind", WithoutInitializing),
                  A_lcl.graph.entries.extent (0));
+
+  // DEEP_COPY REVIEW - DEVICE-TO-DEVICE
   Kokkos::deep_copy (ind, A_lcl.graph.entries);
 
   using offsets_type =
     typename local_graph_type::row_map_type::non_const_type;
   offsets_type ptr (view_alloc ("ptr", WithoutInitializing),
                     A_lcl.graph.row_map.extent (0));
+
+  // DEEP_COPY REVIEW - DEVICE-TO-DEVICE
   Kokkos::deep_copy (ptr, A_lcl.graph.row_map);
 
   using values_type = typename local_matrix_type::values_type;
   values_type val (view_alloc ("val", WithoutInitializing),
                    A_lcl.values.extent (0));
+
+  // DEEP_COPY REVIEW - DEVICE-TO-DEVICE
```

Comment:
Instead of allocating - deep copy three times, we may want to allocate 3 views first and call 3 deep copies. In this way, probably the code does not have to use a system call and it can run without blocking from the system call.